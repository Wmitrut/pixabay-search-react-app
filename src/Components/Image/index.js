const Image = (props) => {
  return (
    <div
      className="flex max-w-xs flex-wrap overflow-hidden  rounded-lg md:w-1/2 lg:w-1/4"
      key={props.index}>
      <div className="w-full max-w-xs overflow-hidden  p-1 md:p-2">
        <img
          alt={props.image.tags}
          loading="lazy"
          className="block h-48 w-full rounded-lg bg-indigo-100 object-cover object-center transition duration-300 ease-in-out hover:scale-110"
          src={props.image.webformatURL}
        />
      </div>
    </div>
  );
};

export default Image;
