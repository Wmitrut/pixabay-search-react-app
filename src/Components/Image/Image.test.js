import renderer from 'react-test-renderer';

import Image from './index';

const image = {
  index: 1,
  tags: 'sun, flower, yellow',
  webFormatUrl: 'https://hosstools.com/wp-content/uploads/2020/10/black-oil-sunflower.jpg',
};

it('Renders correctly', () => {
  const tree = renderer
    .create(<Image image={image} index={image.index} key={image.index} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
