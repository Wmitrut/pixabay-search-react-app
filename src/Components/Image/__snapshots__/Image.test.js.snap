// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="sm:w-1/1 flex max-w-xs flex-wrap overflow-hidden  rounded-lg md:w-1/2 lg:w-1/4"
>
  <div
    className="w-full max-w-xs overflow-hidden  p-1 md:p-2"
  >
    <img
      alt="sun, flower, yellow"
      className="block h-48 w-full rounded-lg bg-indigo-100 object-cover object-center transition duration-300 ease-in-out hover:scale-110"
      loading="lazy"
    />
  </div>
</div>
`;
